driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/syncadapter/SingleResultSubscriber.java

```diff
@@ -58,12 +51,14 @@ T first() {
 
     @Override
     public void onSubscribe(final Subscription s) {
-        s.request(Long.MAX_VALUE);
+        // TODO: Discuss why it's necessary to request more than 1 in order to force a call to onComplete in the case where we're calling
+        // something like FindIterable.first() with a filter by _id.
+        s.request(2);
```

Comment:
@rozza let me know if you have any thoughts about this.

Reply:
I believe its due to the demand driven nature of publisher implementation. I think going forward for  4.0 we can remove that requirement by creating a SinglePublisher implementation that only ever returns a single result. Similar
 to RxJava's Single and the SingleObservable in the Scala driver.

The SingleResultObservableToPublisher could call onComplete straight after onNext but there maybe more work to do on the Observable side.